skpro/regression/linear/_glm.py

```diff
@@ -0,0 +1,444 @@
+        var_weights=None,
+        missing="none",
+        start_params=None,
+        maxiter=100,
+        method="IRLS",
+        tol=1e-8,
+        scale=None,
+        cov_type="nonrobust",
+        cov_kwds=None,
+        use_t=None,
+        full_output=True,
+        disp=False,
+        max_start_irls=3,
+        add_constant=False,
+    ):
+        super().__init__()
+        from statsmodels.genmod.families.family import Gaussian
+
+        self.family = Gaussian()
+        self.offset = offset
+        self.exposure = exposure
+        self.freq_weights = freq_weights
+        self.var_weights = var_weights
+        self.missing = missing
+        self.start_params = start_params
+        self.maxiter = maxiter
+        self.method = method
+        self.tol = tol
+        self.scale = scale
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.use_t = use_t
+        self.full_output = full_output
+        self.disp = disp
+        self.max_start_irls = max_start_irls
+        self.add_constant = add_constant
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            A (n,k) array where n is the number of observations and k is the number
+            of regressors. An intercept is not included by default and should be
+            added by the user (models specified using a formula include an
+            intercept by default). Equivalent to statsmodel's (exog).
+
+        y : pandas DataFrame
+            1d array of the endogenous response variable. This array can be 1d or
+            2d. Binomial family models accept a 2d array  with two columns.
+            If supplied each observation is expected to  be [success, failure].
+            Equivalent to statsmodel's (endog).
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from statsmodels.genmod.generalized_linear_model import GLM
+        from statsmodels.tools import add_constant
+
+        if self.add_constant:
+            X = add_constant(X)
+
+        y_col = y.columns
+
+        glm_estimator = GLM(
+            endog=y,
+            exog=X,
+            family=self.family,
+            offset=self.offset,
+            exposure=self.exposure,
+            freq_weights=self.freq_weights,
+            var_weights=self.var_weights,
+            missing=self.missing,
+        )
+
+        self._estimator = glm_estimator
+
+        fitted_glm_model = glm_estimator.fit(
+            self.start_params,
+            self.maxiter,
+            self.method,
+            self.tol,
+            self.scale,
+            self.cov_type,
+            self.cov_kwds,
+            self.use_t,
+            self.full_output,
+            self.disp,
+            self.max_start_irls,
+        )
+
+        PARAMS_TO_FORWARD = {
```

Comment:
I don't think that the data relatd vars (endog to var_weights) are too interesting.

Either way, you should not overwrite any of the params set in `__init__`.

Reply:
When i ran the tests, i got this result ``` 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-0-ProbaRegressorBasic]': 'PASSED',
 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-0-ProbaRegressorXcolMixIxYnp]': 'PASSED',
 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-0-ProbaRegressorSurvival]': 'PASSED',
 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-1-ProbaRegressorBasic]': 'PASSED',
 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-1-ProbaRegressorXcolMixIxYnp]': 'PASSED',
 'test_fit_does_not_overwrite_hyper_params[GLMRegressor-1-ProbaRegressorSurvival]': 'PASSED',```
 
its likely they passed because i have two family parameters: ```self.family```, and ```self.family_```. Regardless, i have removed ```self.family_```.

Reply:
hm, maybe they end up identical?